firmware/mcal/stm32f767/periph/analog_output.h

```diff
@@ -0,0 +1,36 @@
+/// @author Tamara Xu
+/// @date 2024-10-13
+
+#pragma once
+
+#include <cstdint>
+
+#include "shared/periph/analog_output.h"
+#include "shared/util/mappers/clamper.h"
+#include "stm32f7xx_hal.h"
+
+namespace mcal::stm32f767::periph {
+
+class AnalogOutput : public shared::periph::AnalogOutput {
+public:
+    AnalogOutput(TIM_HandleTypeDef* htim, uint32_t channel)
+        : htim_(htim), channel_(channel) {}
+
+    void SetVoltage (float voltage) override {
+        voltage_ = shared::util::Clamper<float>::Evaluate(voltage, 0, 3.3f);
+        uint32_t pulse = (voltage_ * float(__HAL_TIM_GetAutoreload(htim_))) /
```

Comment:
why do we have a factor of `1/100.0`? I don't think this is correct. If `Autoreload` = 1000, then I need my pulse to be 1000 to set voltage = 3.3V, but this calculation gives 10. Similarly `maxPulse` should be the maximum register value which is a raw integer, not a float

Reply:
You should first calculate `uint32_t maxPulse`, then `pulse = Clamp(voltage / 3.3 * maxPulse, 0, maxPulse)`